portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,31 +14,64 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
-import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+      throws IOException {
+    Query query = new Query("input");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Comment> comments = new ArrayList<>();
+    
+    int count = 0;
+    for (Entity entity : results.asIterable()) {
+    	String name = (String) entity.getProperty("name");
```

Comment:
It's a good idea to use class constants for hard-coded strings and primitive types. It makes it easier to change these and use them multiple times (which you do). It also aids in readability and allows you to add comments to clarify their purpose if necessary.

Note that I wouldn't ask you to add a class constant for something like "application/json;" or "text-input" because that's not something that is controlled/needs to be explained by your code and is very unlikely to change.

example:  http://www.javapractices.com/topic/TopicAction.do?Id=2

Reply:
+1 to this, having a constant for "name" and "comment" and maybe "input" will make it clearer that these strings are intentionally identical between here and the doPost method.